storage/shard.go

```diff
@@ -1682,13 +1685,6 @@ func (s *dbShard) Flush(
 		return s.markFlushStateSuccessOrError(blockStart, err)
 	}
 
-	// No action is necessary therefore we bail out early and there is no need to close.
```

Comment:
mind explicitly setting DeleteIfExists to false in the prepare opts and explain why you choose to do that.

Reply:
Added this.

Reply:
Cheers